packages/grid/_modules_/grid/components/containers/GridRootStyles.ts

```diff
@@ -103,26 +103,40 @@ export const useStyles = makeStyles(
           alignItems: 'center',
         },
         '& .MuiDataGrid-colCellTitleContainer': {
-          textOverflow: 'ellipsis',
-          overflow: 'hidden',
-          whiteSpace: 'nowrap',
-          display: 'inline-flex',
+          display: 'flex',
+          alignItems: 'center',
+          minWidth: 0,
           flex: 1,
-        },
-        '& .MuiDataGrid-colCellNumeric .MuiDataGrid-iconButtonContainer': {
-          paddingRight: 5,
+          whiteSpace: 'nowrap',
         },
         '& .MuiDataGrid-colCellSortable': {
           cursor: 'pointer',
+          '&:hover .MuiDataGrid-sortIcon': {
+            opacity: 1,
+          },
```

Comment:
This looks wrong;

- We don't use opacity 1 in the `<Table>` for the hover unsorted state.
- I was also expecting more **isolation** in the style. Why do we specifically target the styles of `GridColumnUnsortedIcon`?